src/core/ext/filters/client_channel/resolver/xds/xds_resolver.h

```diff
@@ -21,11 +21,36 @@
 
 #include "absl/strings/string_view.h"
 
+#include "src/core/lib/gprpp/ref_counted_ptr.h"
 #include "src/core/lib/gprpp/unique_type_name.h"
 #include "src/core/lib/service_config/service_config_call_data.h"
 
 namespace grpc_core {
 
+// Forward declaration only, implementation is private
+class XdsClusterMap;
+class ClusterState;
+
+class XdsClusterMapAttribute
```

Comment:
It occurs to me that this class can actually be moved into the .cc file.  The only two pieces of code that ever access this are the `XdsConfigSelector` code that sets the attribute and the `ClusterSelectionFilter` code that uses the attribute, and they are both in the .cc file, so it doesn't need to be exported anywhere else.